documentation/src/docs/security/generating-tls-artifacts.md

```diff
@@ -0,0 +1,324 @@
+for instructions.
+
+**Before you Begin:**
+
+The steps described in this document use OpenSSL and Java Keytool to generate TLS artifacts like certificates, keys, 
+Java keystores and Java truststores. Install OpenSSL and Java Development Kit (JDK) on the hosts where the artifacts 
+will be generated.
+
+**Note:**
+* The examples shown in this document use command line arguments to pass all inputs to the command. To pass
+  sensitive command arguments via prompts instead, just exclude the corresponding option (as shown in the second command
+  below). 
+
+  ```bash
+  # Passwords are passed as command line arguments.
+  $ keytool -keystore server01.keystore.jks -alias server01 -validity <validity> -genkey \
+              -storepass <keystore-password> -keypass <key-password> \
+              -dname <distinguished-name> -ext SAN=DNS:<hostname>,
+
+  # Passwords (and some other arguments) are to be entered interactively on the prompt.
+  $ keytool -keystore server01.keystore.jks -alias server01 -genkey
+  ```
+* A weak password `changeit` is used everywhere in this document for easier reading. Be sure to replace it with 
+  a strong and separate password for each file.
+
+## Setting up a Certificate Authority (CA)
+
+If you are planning to use a public CA or an existing private/internal CA, you may skip 
+this part altogether and go directly to the [next section](#generate-tls-certificates-and-other-tls-artifacts-for-the-server).
+
+Here, we'll create a CA in the form of a public/private key pair and a self-signed certificate.
+Later, we'll use this CA certificate/key bundle to sign the servers' certificates.
+
+1. Generate a CA certificate and public/private key pair. 
+
+   ```bash
+   $ openssl req -new -x509 -keyout <ca-private-key-file-path> -out <ca-cert-file-path> -days <validity-in-days> \
+            -subj "<distinguished_name>" \
+            -passout pass:<ca-private-key-password>
+   ```
+
+   Example:
+   ```bash
+   $ openssl req -new -x509 -keyout ca-key.key -out ca-cert.crt -days 365 \
+            -subj "/C=US/ST=Washington/L=Seattle/O=Pravega/OU=CA/CN=Pravega-Stack-CA" \
+            -passout pass:changeit
+   ```
+  
+   The command above will generate the following two PEM-encoded files: 
+   * A file containing the encrypted private key.
+   * A file containing the CA certificate.
+   
+2. Optionally, create a Java truststore containing the CA's certificate. This may be used by client applications if 
+   they configure the truststore using the `javax.net.ssl.trustStore` Java option. 
+   
+   ```bash
+   $ keytool -keystore <java-truststore-file-path> -noprompt \
+             -alias <trusted-cert-entry-alias> \
+             -import -file <ca-cert-file-path> \
+             -storepass <java-truststore-file-password>
+   ```
+   
+   Example:
+   ```bash
+   $ keytool -keystore client.truststore.jks -noprompt \
+             -alias caroot \ 
+             -import -file ca-cert.crt \
+             -storepass changeit
```

Comment:
Is it worth suggesting that private key and truststore use different passwords? Is that even preferable?

Reply:
An earlier line 85 says that: `A weak password `changeit` is used everywhere in this document for easier reading. Be sure to replace it with a strong and separate password for each file.`

Yes, each password should be different. Reusing a password is usually considered a bad practice.